test/testdata/rewriter/nesting.rb

```diff
@@ -0,0 +1,57 @@
+# typed: strict
+
+class A
+  extend T::Sig
+  sig {void}
+  def outer
+    sig {void}
```

Comment:
I'm honestly a little surprised that `sig` is not a NoMethodError right here

Reply:
The rewriting here goes from
```ruby
class A
  extend T::Sig
  sig {void}
  def outer
    sig {void}
    def inner; end
  end
end
```
to
```ruby
class A
  extend T::Sig
  sig {void}
  def outer; end
  sig {void}
  def inner; end
end
```
so `sig` is still being called on the class and is also sticking with its associated `def`.